index.d.ts

```diff
@@ -0,0 +1,6 @@
+export as namespace timekeeper;
```

Comment:
Would you mind explaining how this works?

Just on the top of my head, I would have written it like this, but maybe your way is better ☺️ 

```typescript
interface Timekeeper {
  freeze(): void
  travel(date: Date): void
  reset(): void
  isKeepingTime(): boolean
}

declare const timekeeper: Timekeeper

export = timekeeper
```

Reply:
My grasp of TS is relatively primitive, but here goes...

It seems like this first line is necessary for when the library is used as a browser global (see the comment above a similar line in the [module.d.ts example code](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html)). This mechanism provides the name of the field appended to the global object.

The `export = ` syntax is needed for module-functions and module-classes, since these are providing a default export. That's not the case here so I've directly exported the function signatures.

Reply:
> It seems like this first line is necessary for when the library is used as a browser global 

Ahh, of course, I was thinking about this from a purely Node.js perspective...

> [...] since these are providing a default export.

Actually, I think it's more about if the file is a CommonJS style module or an ES-module. In this case, we are assigning to `module.exports = ` in the source code, so I believe that `export =` is correct in the Node.js case.

DefinitelyTyped's guidelines is to follow the source of the library, briefly mentioned here: https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

Reply:
Hmm, after reading this: https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#support-for-umd-module-definitions

I think that the following would serve us well:

```typescript
export as namespace timekeeper

interface Timekeeper {
  freeze(): void
  travel(date: Date): void
  reset(): void
  isKeepingTime(): boolean
}

declare const timekeeper: Timekeeper

export = timekeeper
```

Reply:
Works for me. I'll push an update to this branch. :+1:

Reply:
Pushed.